web_renderer/src/server.ts

```diff
@@ -0,0 +1,72 @@
+import express, { Request, Response } from 'express';
+import { SessionId } from './common';
+import { Session } from './session';
+import { constants as HttpConstants } from 'http2';
+import { randomUUID } from 'crypto';
+import { GetFrameRequest, NewSessionRequest } from './schemas';
+import { sessions } from './state';
+
+interface ErrorResponse {
+    error: string
+}
+
+type NewSessionResponse =
+    { session_id: SessionId }
+    | ErrorResponse;
+
+type GetFrameResponse =
+    Buffer
+    | ErrorResponse;
+
+    
+const app = express();
+
+app.use(express.json());
+
+app.post("/new_session", (req: Request, res: Response<NewSessionResponse>) => {
+    try {
+        const data = NewSessionRequest.parse(req.body);
+        console.log(`Starting rendering for ${data.url}`);
+        const session_id = randomUUID();
+        const session = new Session(data.url, data.resolution);
+        sessions.set(session_id, session);
+        session.run();
+
+        res
+            .status(HttpConstants.HTTP_STATUS_CREATED)
+            .send({
+                session_id: session_id
+            });
+    } catch (err) {
+        res
+            .status(HttpConstants.HTTP_STATUS_BAD_REQUEST)
```

Comment:
let's leave for now, but usually you are just throwing an error and add middleware that will convert it to the response